utils/build-script-impl

```diff
@@ -641,7 +641,10 @@ if [[ "$SKIP_TEST_WATCHOS" ]] ; then
     SKIP_TEST_WATCHOS_SIMULATOR=1
 fi
 
-if [[ "${CMAKE_GENERATOR}" == "Ninja" ]] && [[ -z "$(which ninja)" ]] ; then
+# Some Linux distributions, e.g. Fedora, use 'ninja-build' instead of
+# 'ninja' as binary name. Check for both.
+if [[ "${CMAKE_GENERATOR}" == "Ninja" ]] && [[ -z "$(which ninja)" ]] &&
+   [[ -z "$(which ninja-build)" ]] ; then
```

Comment:
It works because in both cases the return value is not empty (which is what is being tested) therefore the last two conditions always yield true. The previous implementation was also wrong.

`which` has a 0 exit code for successful find and another when it doesn't so a correct patch should be: (according to this http://blog.sanctum.geek.nz/testing-exit-values-bash/)

```
if [[ "${CMAKE_GENERATOR}" == "Ninja" ]] && (( [[ "$(which ninja)" ]] ||
   [[ -z "$(which ninja-build)" ]] )) ; then
```

That is testing that:
1.- CMAKE_GENERATOR equals to Ninja
2 .- ninja OR ninja-build exist in the system

I did a small test in OS X and it works so it should as well in Fedora but you should check in your environment